kernel/nano_core/src/uefi.rs

```diff
@@ -0,0 +1,67 @@
+use crate::{early_setup, nano_core, try_exit};
+use boot_info::uefi::STACK_SIZE;
+use bootloader_api::{config::Mapping, BootloaderConfig};
+use core::arch::asm;
+use memory::VirtualAddress;
+
+#[used]
+#[link_section = ".bootloader-config"]
+pub static __BOOTLOADER_CONFIG: [u8; BootloaderConfig::SERIALIZED_LEN] = {
+    let mut config = BootloaderConfig::new_default();
+    config.mappings.physical_memory = Some(Mapping::Dynamic);
+    config.mappings.page_table_recursive =
+        Some(Mapping::FixedAddress(0o177777_776_000_000_000_0000));
+    config.kernel_stack_size = STACK_SIZE as u64;
+    config.serialize()
+};
+
+#[naked]
+#[no_mangle]
+#[link_section = ".init.text"]
+pub unsafe extern "C" fn _start(boot_info: &'static mut bootloader_api::BootInfo) {
```

Comment:
any particular reason why this is static mut? 
Also, how is it a rust-level reference and not a raw pointer if this is the first time we're entering Rust code? I understand they are interchangeable but it still conveys a misleading calling context.

Reply:
> any particular reason why this is static mut?

Changed

> Also, how is it a rust-level reference and not a raw pointer if this is the first time we're entering Rust code?

The bootloader runs rust code and directly calls `_start` with a Rust reference.

From <https://github.com/rust-osdev/bootloader>:
> Define an entry point function with the signature `fn kernel_main(boot_info: &'static mut bootloader_api::BootInfo) -> !`

Reply:
ok i see, didn't realize we made the transition from Rust -> asm -> Rust again. Works for me.